devito/finite_differences/coefficients.py

```diff
@@ -220,24 +231,26 @@ def generate_subs(i):
 def default_rules(obj, functions):
 
     def generate_subs(deriv_order, function, dim):
+        retrieved_dim = retrieve_dimension(dim)[0]
```

Comment:
Looks to me like the `[0]` should be in the function itself and not here?

Reply:
But also, the naming still needs changing, e.g. `dim = retrieve_dim(index)` since `dim` is not always a `dim`.

Reply:
Made the change to the naming